test_pull_requests

```diff
@@ -935,6 +969,91 @@ popd
+
+      valid_comment_exists = false
+      flake_label = settings['flake_identification']['label']
+      flake_repo = settings['flake_identification']['repo']
+      org_repo = %{#{Properties['github_user']}/#{flake_repo}}
+      flake_label_query = CGI.escape("label:#{flake_label}")
+      issue_link = %{https://github.com/#{org_repo}/issues?q=#{flake_label_query}}
+      reason = %{No comment was found that meets the flake identification requirements. A correctly formatted comment should include a link to the failed Jenkins job and at least one link to a GitHub issue in the [`#{org_repo}` repository](#{issue_link}) with the `#{flake_label}` label. An example of a comment with correct syntax would look like:
+> @#{Properties['bot_github_user']}, the [last build](#{build_url}) failed due to the following flakes:
+>  - https://github.com/#{org_repo}/issues/0
+>  - https://github.com/#{org_repo}/issues/00}
+
+      comments = sort_comments(comments)
+      comments.each do |comment|
+        comment_matches_syntax = comment['body'] =~ /#{Regexp.quote(build_url)}/ && comment['body'] =~ general_issue_spec
+        # we can't allow the bot to be a valid author, or we would pick up
+        # comments where the bot explains the flake syntax to GitHub users
+        comment_author_valid = comment['user']['login'] != Properties['bot_github_user']
+        if comment_matches_syntax && comment_author_valid
+          # if we find a comment that matches the flake job syntax from someone
+          # trusted, we need to validate the comment's flake issue list
+          bad_issue_links = []
+          comment['body'].scan(general_issue_spec) do |issue|
+            if issue =~ flake_issue_spec
+              issue_url = $1
+              issue_id = $2.to_i
+              if !issue_has_test_flake_label(issue_id, flake_repo, flake_label)
+                # the issue does not have the labels necessary, so it is invalid
+                bad_issue_links << issue_url
+              end
+            else
+              # we had a GitHub issue link, but it didn't
+              # match the org or repo that we wanted
+              bad_issue_links << issue
+            end
+          end
+
+          if bad_issue_links.length == 0
+            valid_comment_exists = true
+            break
+          else
+            # any invalid issue links mean that the comment is invalid
+            reason = %{A [comment](#{comment["html_url"]}) was found that referenced the correct Jenkins job URL, but some listed issue links were invalid. Ensure all linked issues are in the [`#{org_repo}` repository](#{issue_link}) and have the `#{flake_label}` label. Invalid links are:
+ - #{bad_issue_links.join("\n - ")}}
+          end
+        end
+      end
+
+      [valid_comment_exists, reason]
+    end
+
+    # issue_has_test_flake_labels determines if the issue at the
+    # repository and issue ID has a test flake label
+    def issue_has_test_flake_label(issue_id, repo, label)
```

Comment:
method should end in a ?

method says it is looking for a flake label but it looks perfectly generic.

method should probably be called issue_has_label?